### PR TITLE
Lane-wise shifts

### PR DESCRIPTION
**Pull request by penzn**
Introduce equivalents of lane-wise (or 'logical") shifts, present in popular SIMD instruction sets.

### CODE REVIEW

File: proposals/flexible-vectors/FlexibleVectors.md [hunk 1]
```diff
@@ -141,8 +141,53 @@ def S.replace_lane(a, i, x):
...
 
+### Shuffles
 
-### Integer arithmetic
+#### Left lane-wise shift by scalar
+
+* `vec.i8.lane_shift_left(a: vec.i8, x: i32) -> vec.i8`
+* `vec.i16.lane_shift_left(a: vec.i16, x: i32) -> vec.i16`
+* `vec.i32.lane_shift_left(a: vec.i32, x: i32) -> vec.i32`
+* `vec.i64.lane_shift_left(a: vec.i64, x: i32) -> vec.i64`
```
Comment (abrown) [inline proposals/flexible-vectors/FlexibleVectors.md]: Random drive-by comment: you don't want to use the same nomenclature as the 128-bit spec? e.g. `shl/shr_u/shr_s`?

Comment (penzn) [reply proposals/flexible-vectors/FlexibleVectors.md]: Those are already in - this change introduces "packed shift" instructions (for example, `psll*`/`psrl*` instructions on x86 and `lsl`/`lsr` on Arm). I do think we need a better name here though 😄

Comment (abrown) [reply proposals/flexible-vectors/FlexibleVectors.md]: Oh, it's shifting lanes, not bits!

> for example, psll*/psrl* instructions on x86 and lsl/lsr on Arm

But `shl` already uses `PSLL{W/D/Q}`, e.g., so do you mean that this would use like `PSLLDQ`/`VPSLLDQ`?

Comment (abrown) [reply proposals/flexible-vectors/FlexibleVectors.md]: :+1: for a more clear name 😆

Comment (penzn) [reply proposals/flexible-vectors/FlexibleVectors.md]: > But `shl` already uses `PSLL{W/D/Q}`, e.g., so do you mean that this would use like `PSLLDQ`/`VPSLLDQ`?

Ah, you are right, it is `PS[LR]LDQ`. I am not sure what would be a better name, maybe `lsh[lr]`?

Edit: for some weird reason I started a "review" and did not notice that the comment was hanging in pending status for a couple of days.

**Comment by penzn** [comment]
By the way, if anybody else wants to join in reviewing PRs to this proposal, it would be delightful!

The goal at the moment is to provide operations equivalent to operations in the simd proposal, which exception of shuffle (where there would not be an exact match). This PR provided one of the shuffle-themed operations, and there will be more later.

CC @sampsyo, @zeux, and @ngzhian
